### DISCUSSION TITLE
Driver I2C for SH1122 not working with U8x8lib

### DISCUSSION THREAD

**Discussion by spiderjuka** [olikraus/u8g2#1456] [Q&A]
Hello,

Can you support me how can I use display SH1122 under I2C with library U8x8lib?

It doesn't work. The screen does not initialize properly - there are a lot of white pixels on the screen.
It working with library U8g2lib, but is too slow.

Thank you.

**Comment by olikraus** [comment]
The SH1122 driver is not supported by U8x8. Did you use u8g2 with hardware I2C? It should be fast enough.

**Reply by spiderjuka** [reply]
Yes I used but unfortunatelly I2C (hardware) is still not fast enough.
I use that display with ESP12E.
I will try buy display with SPI interface.
How do yo think can it be better then I2C?

Thank you.

**Reply by olikraus** [reply]
Spi usually is faster.
